Sources/ContentfulPersistence/SynchronizationManager.swift

```diff
@@ -123,6 +128,68 @@ public class SynchronizationManager: PersistenceIntegration {
         }
         self.localeCodes = localeCodes
     }
+    
+    /// Synchronizes data using the provided localization scheme, performing a follow-up sync with all localizations if required.
+    /// - Parameters:
+    ///   - syncSpacePersistable: The type conforming to `SyncSpacePersistable` used for handling synchronization data.
+    ///   - initialLocalizationScheme: The localization scheme to use for the initial synchronization phase.
+    ///     This must not be `.all`.
+    ///   - limit: An optional limit for the number of records to synchronize in one batch. Defaults to `nil`, indicating no limit.
+    ///   - dbVersion: The database version targeted for synchronization. Defaults to the application's default database version.
+    ///   - onInitialCompletion: A closure called upon completion of the initial synchronization phase, with a `Result` containing either the `SyncSpace` or an error.
```

Comment:
Maybe we could have just a generic "on locale sync completed" with parameter that is a locale code and let third party devs decide what action they want to take after specific locale is synced rather than having 2 completion methods?